### ISSUE TITLE
Help.md contains Obsolete Links

### ISSUE THREAD

**Issue by AmanNegi** [dart-lang/pub-dev#6374]
# Issue

The `Help.md` under `pub.dev/doc` contains Obsolete unreachable links. All the links provided are unreachable and should be replaced with appropriate(reachable) links. So that any beginner contributor can access these resources.

# Screenshot
![image](https://user-images.githubusercontent.com/37607224/214838413-39a236a8-40dd-44cd-810b-c227d6b34c06.png)

**Comment by AmanNegi** [comment]
Hey there, I can fix this issue can you please provide me the updated links? I would love to contribute to this issue.

**Comment by isoos** [comment]
@AmanNegi: `help.md` is the file that we display on the https://pub.dev/help page, it is not meant to be linking anything in the pub-dev repository. For contributing, the following page is a good start:
https://github.com/dart-lang/pub-dev/blob/master/doc/development.md

Please note, some features may sound good at first, but we may not want to work on them or accept contributions on them, because they may be blocked by something else (e.g. design or other feature that would make it obsolete). So before contributing a larger chunk, please reach out e.g. here in the issues what you are planning to do and what direction it should be going...

**Comment by AmanNegi** [comment]
I am really sorry I am new to this repository, so I had no idea that it was used in `pub.dev`. 

Also I am facing issues setting up `pub.dev` on my local system, I am following the steps mentioned in development.md to create fake server however I am getting this [error](https://pastebin.com/6eK6DsFT).

Can you suggest some resources that may help me in this.

**Comment by isoos** [comment]
This seems to be a Windows-related error we don't normally see, as none of us is developing on Windows directly, maybe on WSL2. I think we had a similar report on`pana` a few months ago, but it was resolved.

It seems to happen when the analysis is triggered. Could you please try something? Checkout [pana](https://github.com/dart-lang/pana) and run it locally on one of the Flutter packages that you've included in your test profile.

**Comment by AmanNegi** [comment]
I tried running `pana` on the packages, and it does run successfully. I am not sure where Dart is saving its packages. I found some dart packages in this path `C:\Users\Aster\AppData\Local\Pub\Cache`. I tried adding `google_fonts` using `dart pub get google_fonts` however it is not reflected in the path. I am not sure where it is saving the files as a cache.

Irrespective the error is still the same. How was it solved earlier, any special remarks on that?

**Comment by isoos** [comment]
> How was it solved earlier, any special remarks on that?

It was in the cleanup phase of the analysis, and IIRC the path had spaces in it. We did modify a few parts of the code around it, and there was no specific fix that I recall.

>  I found some dart packages in this path C:\Users\Aster\AppData\Local\Pub\Cache

That seems to be the regular `PUB_CACHE` directory. `pana` analysis stores has its own environment with its own `PUB_CACHE`, so it is probably not related (hence the different paths, e.g. the analysis is inside a temp folder by design).

To unblock you, you may try to use the fake analyzer in the init-data-file command, so that we skip pana analysis and use just some placeholder values.

**Comment by AmanNegi** [comment]
I am using fake analysis:
`dart bin/fake_server.dart init-data-file --test-profile=test.yaml --analysis=fake --data-file=dev-data-file.json`

I also work on Flutter, could that be an issue of the problem?

**Comment by isoos** [comment]
> I also work on Flutter, could that be an issue of the problem?

Maybe. We do have Flutter in path, but use only Dart SDK to develop/run the project.

**Comment by AmanNegi** [comment]
Yeah, I am only using Dart SDK only but it is downloaded/present inside the Flutter directory only `C:\src\flutter\bin\cache\dart-sdk\bin`. Could this be causing the issue?

**Comment by isoos** [comment]
It is possible, but unlikely. I'm unsure what the root cause could be.

**Comment by isoos** [comment]
@AmanNegi: maybe a thing to debug: run the tests we have in `app/test`, and let's see if any or which break.

**Comment by AmanNegi** [comment]
If you think it is related to windows only, I could try it on Linux. If I am right, I just need the Dart SDK and I'll be good to go or I need to install other tools as well?

> PS: I am trying to run the tests they seem to run just fine.

**Comment by AmanNegi** [comment]
Update: Received several similar errors while running tests as well.

`PathNotFoundException: Directory listing failed, path='C:\Users\Aster\AppData\Local\Temp\dart-tempdir269c53ce\pub-cache\hosted\pub.dartlang.org\*`

**Comment by isoos** [comment]
Which tests are failing? Can you also include a stack trace?

**Comment by AmanNegi** [comment]
The stack trace is very long however I was able to get these starting errors from the terminal https://pastebin.com/wD7ixw7A

**Comment by AmanNegi** [comment]
In Ubuntu also I am receiving a similar error: https://pastebin.com/BSqxV4NZ

What Steps I did:

1. Installed Dart according to the [Documentation](https://dart.dev/get-dart),
2. Cloned the repository `pub-dev` in my system.
3. Ran `dart pub get`.
4. In terminal went to `pub-dev/app` and executed the following command: 
`dart bin/fake_server.dart init-data-file --test-profile=test.yaml --analysis=fake --data-file=dev-data-file.jsonl`

Am I missing something or doing something wrong?

**Comment by isoos** [comment]
@AmanNegi: I wasn't able to reproduce the issue on windows, but we should be able to narrow down the one on linux/ubuntu. A difference there could be uninitialized packages in the pkg/ directory. Could you run `pub get` in all of them? To make it automatic, you could do the following in the `pub-dev` root:

`dart pub global activate mono_repo` (this install `package:mono_repo` in the global scope)
`dart pub global run mono_repo pub get` (this traverses the project directory and runs `pub get` in each package)

And then run the init-data-file again...

**Comment by AmanNegi** [comment]
I tried that but the error is still the same: `PathNotFoundException: Directory listing failed, path = '/tmp/dart-tempdirLMPWLC/pub-cache/hosted/pub.dartlang.org/' (OS Error: No such file or directory, errno = 2)`

Does it have something to do with redis?

**Comment by AmanNegi** [comment]
I will create a new issue for this, so that the details can be organized, in once place.
